### ISSUE TITLE
Fix request: false positive in AvoidUnusedAssignments

### ISSUE THREAD

**Issue by techie-jyoti** [jborgers/PMD-jPinpoint-rules#229]
For DB entity classes , It is giving false positive issue  "Avoid assignments for which the assigned value is never used'

Example for below class , it is giving issue for localDate and uuid attributes, stating assigned variable is never used (in Constructor or in any other methods if any) 

```
@Entity
public class Customer {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long Id;


    @NotNull
    private String uuid;

    private LocalDateTime localDate;

    public Customer() {
        date = now();
        uuid = UUID.randomUUID().toString();
    }
}
```

**Comment by techie-jyoti** [comment]
Hi , I am able to reproduce using below sample code 


import static java.time.LocalDateTime.now;

import java.time.LocalDateTime;
import java.util.UUID;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.validation.constraints.NotNull;

@Entity
public class Customer {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long Id;


    @NotNull
    private String uuid;

    private LocalDateTime date;

    public Customer() {
        date = now();
        uuid = UUID.randomUUID().toString();
    }

    public void Register(){
        date = now();
        uuid = UUID.randomUUID().toString();
    }
}


<img width="956" alt="image" src="https://user-images.githubusercontent.com/106815870/203074807-7f539cf5-9e1c-4ef0-b4b2-7aa826f0c96a.png">

**Comment by jborgers** [comment]
Simplified test case:
```
public class Foo {

    private String str1;

    public Foo() {
        str1 = "now"; // false positive
        String str2 = "now"; // false positive
    }

    public void when(){
        str1 = "now";
        String str2 = "now"; // bad
        str2 = "now"; 
    }
}
```

**Comment by jborgers** [comment]
Fixed

**Comment by koosg** [comment]
The following also gives a false positive. Possibly because of using `+=`
```
public static boolean elfProef(final String accountNumber) {
    int sum = 0;
    final int length = accountNumber.length();
    for (int i = 0; i < length; ++i) {
        sum += Character.digit(accountNumber.charAt(i), RADIX) * (length - i);
    }
    return sum % DIVIDER == 0;
}
```
It complains that the assignment to `int sum=0` should be avoided...
